atomicfu/build.gradle.kts

```
@@ -0,0 +1,382 @@
+            dependencies {
+                implementation("org.jetbrains.kotlin:kotlin-reflect")
+                implementation("org.jetbrains.kotlin:kotlin-test")
+                implementation("org.jetbrains.kotlin:kotlin-test-junit")
+                implementation(libs.junit.junit)
+            }
+        }
+    }
+}
+
+// Support of all non-deprecated targets from the official tier list: https://kotlinlang.org/docs/native-target-support.html
+kotlin {
+    // Tier 1
+    macosX64()
+    macosArm64()
+    iosSimulatorArm64()
+    iosX64()
+
+    // Tier 2
+    linuxX64()
+    linuxArm64()
+    watchosSimulatorArm64()
+    watchosX64()
+    watchosArm32()
+    watchosArm64()
+    tvosSimulatorArm64()
+    tvosX64()
+    tvosArm64()
+    iosArm64()
+
+    // Tier 3
+    androidNativeArm32()
+    androidNativeArm64()
+    androidNativeX86()
+    androidNativeX64()
+    mingwX64()
+    watchosDeviceArm64()
+
+    @OptIn(ExperimentalKotlinGradlePluginApi::class)
+    applyDefaultHierarchyTemplate {
+        group("native") {
+            group("nativeUnixLike") {
+                withLinux()
+                withApple()
+            }
+        }
+        group("androidNative32Bit") {
+            withAndroidNativeX86()
+            withCompilations { compilation ->
+                (compilation.target as? KotlinNativeTarget)?.konanTarget?.name == "android_arm32"
+            }
+        }
+        group("androidNative64Bit") {
+            withAndroidNativeArm64()
+            withAndroidNativeX64()
+        }
+
+    }
+
+    sourceSets {
+        val nativeUnixLikeMain by getting {
+            kotlin.srcDir("src/nativeUnixLikeMain/kotlin")
+            dependsOn(nativeMain.get())
+        }
+
+        val androidNative32BitMain by getting {
+            kotlin.srcDir("src/androidNative32BitMain/kotlin")
+            dependsOn(nativeMain.get())
+        }
+
+        val androidNative64BitMain by getting {
+            kotlin.srcDir("src/androidNative64BitMain/kotlin")
+            dependsOn(nativeMain.get())
+        }
+
+        val androidNative32BitTest by getting {
+            kotlin.srcDir("src/androidNative32BitTest/kotlin")
+            dependsOn(nativeTest.get())
+        }
+
+        val androidNative64BitTest by getting {
+            kotlin.srcDir("src/androidNative64BitTest/kotlin")
+            dependsOn(nativeTest.get())
+        }
+
+    }
+
+    // atomicfu-cinterop-interop.klib with an empty interop.def file will still be published for compatibility reasons (see KT-68411)
+    // This block can be removed when this issue in K/N compiler is resolved: KT-60874
+    targets.withType(KotlinNativeTarget::class).configureEach {
+        compilations.configureEach {
+            cinterops {
+                val interop by creating {
+                    definitionFile.set(project.file("src/nativeInterop/cinterop/interop.def"))
+                }
+            }
+        }
+    }
+}
+
+val transformer by configurations.creating
```

```suggestion
val transformer: Configuration by configurations.creating
```

suggestion: specific type to avoid "Declaration has type inferred from a platform call..." IntelliJ warning.